config/application.rb

```diff
@@ -263,12 +286,7 @@ def try_config_for(*args)
       config.three_scale.oauth2 = config_for(:oauth2)
     end
 
-    initializer :log_formatter, after: :initialize_logger do
-      config.log_formatter = System::ErrorReporting::LogFormatter.new
-      (config.logger || Rails.logger).formatter = config.log_formatter
-    end
-
-    config.paperclip_defaults = {
+    Rails.application.config.paperclip_defaults = {
```

Comment:
just wondering why Paperclip is different, and the config is accessed through `Rails.application`

Reply:
Because I moved this inside a `config.to_prepare` block, where `config` is out of scope, to fix the autoloader. Later, I found a problem and had to move it out again([commit](https://github.com/3scale/porta/commit/92f6bbffa9f72d7b6cad4c8ecdc00eae09930b4b)). And forgot to remove the `Rails.application` part. Fixed.

Reply:
Speaking of which... Why some many initializers are wrapped into `config.to_prepare` in the first place? Is that required?

From reading about this method, it seems that it's to ensure that the config is reloaded on every request (for dev). But again - do we need this?... :thinking: 

Was this to silence the warnings? I've seen initializers may cause warnings like:
```
DEPRECATION WARNING: Initialization autoloaded the constant XXX
```

Reply:
`config.to_prepare` silences that deprecation warning, indeed. And it's required for two reasons:

1. Without it, we'd could experience weird errors at any moment after making changes to `*.rb` files in our dev environments. In practice, we would need to restart the server after every change to avoid the errors.
2. In rails 7 the deprecation warning has been deleted and an error is returned instead. So in our case porta wouldn't even start in rails 7.

The blocks under `config.to_prepare` are not called for every request, but for every Zeitwerk reload. That only happens when a request is received **and** there are changes in any `.rb` file. So only the first request after code changed triggers the reload.

Reply:
Thanks for the explanation!